engine/api/bootstrap/bootstrap.go

```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
+
+	pip, err := pipeline.LoadPipeline(db, p.ProjKey, p.PipName, true)
+	if err != nil {
+		return sdk.WrapError(err, "unable to load pipeline")
+	}
+
+	for _, s := range pip.Stages {
+		for _, j := range s.Jobs {
+			var migrateJob bool
+			for _, a := range j.Action.Actions {
+				if a.Name == DEPRECATEDGitClone {
+					log.Info("MigrateActionDEPRECATEDGitClone> Migrate %s/%s/%s(%d)", p.ProjKey, p.PipName, j.Action.Name, j.Action.ID)
+					migrateJob = true
+					break
+				}
+			}
+			if migrateJob {
+				if err := MigrateActionDEPRECATEDGitCloneJob(db, store, p.ProjKey, p.AppName, j); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+var (
+	originalGitClone *sdk.Action
+	anAdminID        int64
+)
+
+// MigrateActionDEPRECATEDGitCloneJob is the unitary function
+func MigrateActionDEPRECATEDGitCloneJob(db gorp.SqlExecutor, store cache.Store, pkey, appName string, j sdk.Job) error {
+	mapReplacement := make(map[int]sdk.Action)
+
+	var err error
+	//Load the builtin gitclone action is needed
+	if originalGitClone == nil {
+		originalGitClone, err = action.LoadPublicAction(db, sdk.GitCloneAction)
+		if err != nil {
+			return err
+		}
+	}
+
+	//Load the first admin we can
+	if anAdminID == 0 {
+		users, err := user.LoadUsers(db)
+		if err != nil {
+			return err
+		}
+		for _, u := range users {
+			if u.Admin {
+				anAdminID = u.ID
+				break
+			}
+		}
+	}
+
+	//Load the project
+	proj, err := project.Load(db, store, pkey, nil, project.LoadOptions.WithKeys)
+	if err != nil {
+		return err
+	}
+	projKeys := proj.SSHKeys()
+
+	//Load the application
+	log.Debug("load application %s", appName)
+	app, err := application.LoadByName(db, store, pkey, appName, nil, application.LoadOptions.WithKeys)
+	if err != nil {
+		return err
+	}
+	appKeys := app.SSHKeys()
+
+	//Check all the steps of the job
+	for i := range j.Action.Actions {
+		step := &j.Action.Actions[i]
+		log.Debug("CheckJob> Checking step %s", step.Name)
+
+		if step.Name == DEPRECATEDGitClone {
+			//Migrate this step
+			url := sdk.ParameterFind(&step.Parameters, "url")
+			directory := sdk.ParameterFind(&step.Parameters, "directory")
+			branch := sdk.ParameterFind(&step.Parameters, "branch")
+			commit := sdk.ParameterFind(&step.Parameters, "commit")
+
+			newGitClone := sdk.Action{
+				Name:       sdk.GitCloneAction,
+				Enabled:    true,
+				Type:       sdk.DefaultAction,
+				Parameters: originalGitClone.Parameters,
+			}
+
+			//Keep the old parameters
+			sdk.ParameterFind(&newGitClone.Parameters, "url").Value = url.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "directory").Value = directory.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "branch").Value = branch.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "commit").Value = commit.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "user").Value = ""
+			sdk.ParameterFind(&newGitClone.Parameters, "password").Value = ""
+
+			//If there is an application key or a project key, use it
```

Comment:
if url.value == git.http_url do nothing?

Reply:
well i think we can't solve all the problems; we will have to check manually